sourcegraph/sourcegraph
# Code Insights: Implement new "Add or remove insights" view

Closes https://github.com/sourcegraph/sourcegraph/issues/46291

This PR adds new add or remove insights UI. The new UI has 
- **New visual styles**. We use more available space on the screen that show more data at once. It became more closer to other similar views (like Fuzzy Finder UI)
- Fully keyboard accessible UI. Prior to this PR the old view didn't have any keyboard navigation short cuts. Thanks to https://github.com/sourcegraph/sourcegraph/pull/46362 MultiCombobox UI we do have now a full fledged keyboard navigation. 
- **Performance improvements** First of all there is no loading state anymore. We read all needed data from the cache without unnecessary requests at the beginning. The new view supports smart pagination + search over insights by its title and series label.

| Before | After |
| ---------- | --------- |
| <img width="1065" alt="Screenshot 2023-01-16 at 18 37 32" src="https://user-images.githubusercontent.com/18492575/212769415-4b09d06c-7dba-4808-b7cd-d98b50452b8c.png"> | <img width="1065" alt="Screenshot 2023-01-16 at 18 37 12" src="https://user-images.githubusercontent.com/18492575/212769420-ab994920-3251-43ae-89a4-5ae373d562de.png"> |

## Test plan
- Go to the dashboard and click add or remove insights button 
- Add and remove some insights with new UI
- Click save and check that all insights that you remove have been removed and all that you added have been added to the dashboard.



## App preview:

- [Web](https://sg-web-vk-add-new-add-insight-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
